core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorTable.java

```diff
@@ -27,6 +29,18 @@
  * @see SqlValidatorCatalogReader
  */
 public interface SqlValidatorTable {
+
+  /**
+   * Holds a RexBuilder. Insulates SqlValidator implementations from Rex dependencies.
```

Comment:
After the (upcoming) addendum to CALCITE-1666, we can remove `RexBuilder` completely from `SqlValidatorImpl`. But I'm not sure if this is much of an improvement since it just moves the `RexBuilder` dependency from `SqlValidatorImpl` to `SqlValidatorTable`.